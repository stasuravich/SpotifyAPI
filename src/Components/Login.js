import React, {useState} from 'react';
import '../Css/Login.css';
import logo from '../logos/adrian-korte-5gn2soeAc40-unsplash.jpg';

const Login =props=>{

  const [token, setToken]=useState();


  const submit=e =>{
    e.preventDefault();
    if(token){
      props.setLoggedIn(token);
    }
    else{
      props.setWrong(true);
    }
  }


  return (
    <div className="Login">
      <img src={logo} alt ="Logo" className="Logo"/>
      <h1>Login</h1>
      <form onSubmit = {e => submit(e)}>
        <label className="TokenLabel">Enter spotify token below (get it from {" "}
          <a href="https://beta.developer.spotify.com/documentation/web-playback-sdk/quick-start/#authenticating-with-spotify">
            here
          </a>):
        </label>
        <br/>
        <input className="TokenInput" type="text" placeholder="spotify token" onChange={e=> setToken(e.target.value)} defaultValue =''/>
        <br/><br/>
        <button type = "submit" >Submit</button>
      </form>
      {props.wrong &&
        <div className="Wrong"> Invalid token, please try again</div>
      }
    </div>
  );
}

export default Login;
