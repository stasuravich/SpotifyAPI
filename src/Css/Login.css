.Login{
  text-align: center;
}
.Logo{
  padding-top: 25px;
  width: 15%;
}

.TokenLabel{
  display: block;
  font-size: 20px;
}
.TokenInput{
  width: 300px;
}

.Wrong{
  color: red;
}
