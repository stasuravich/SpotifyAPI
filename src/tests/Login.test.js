import React from 'react';
import Login from '../Components/Login';
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";

describe("rendering components", ()=> {
  it("renders App without crashing", ()=>{
    shallow(<Login/>);
  });
  it("renders App component header without crashing", ()=>{
    const wrapper = shallow(<Login/>);
    const header=(<h1>Login</h1>);
    expect(wrapper.contains(header)).toEqual(true);
  })
  it("renders button", ()=>{
    const wrapper = shallow(<Login/>);
    const label=wrapper.find("#submit-button").text();
    expect(label).toEqual("Submit");
  })
})

describe("snapshots", () => {
  it("Login snapshots", () => {
    const tree= shallow(<Login/>);
    expect(toJson(tree)).toMatchSnapshot();
  });
})
