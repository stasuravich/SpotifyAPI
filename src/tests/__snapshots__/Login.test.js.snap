// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Login snapshots 1`] = `
<div
  className="Login"
>
  <img
    alt="Logo"
    className="Logo"
    src="adrian-korte-5gn2soeAc40-unsplash.jpg"
  />
  <h1>
    Login
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      className="TokenLabel"
    >
      Enter spotify token below (get it from 
       
      <a
        href="https://beta.developer.spotify.com/documentation/web-playback-sdk/quick-start/#authenticating-with-spotify"
      >
        here
      </a>
      ):
    </label>
    <br />
    <input
      className="TokenInput"
      defaultValue=""
      onChange={[Function]}
      placeholder="spotify token"
      type="text"
    />
    <br />
    <br />
    <button
      id="submit-button"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
